### ISSUE TITLE
Content is not honored if it points to a file under Output

### ISSUE THREAD

**Issue by livarcocc** [dotnet/sdk#588]
For the project below, the runtimeconfig.json is not packaged because it is a file under OutputPath, which gets excluded from Content.

My worry here is that once we take the new CLI, tools (ASP.NET) are going to be broken and are going to have to do “hacks” to get around this issue.
 
This also may seem somewhat arbitrary to users. They won’t be able to add any Content if that Content is coming from the output. I can see people complaining, not understanding and filing issues on us for this, as there is no real way to find out why this is happening.

```
<Project Sdk="Microsoft.NET.Sdk" ToolsVersion="15.0">
 
  <PropertyGroup>
    <OutputType>Exe</OutputType>
    <TargetFramework>netcoreapp1.0</TargetFramework>
    <GeneratedPackageId>random-name</GeneratedPackageId>
    <PackageId>something</PackageId>
    <AssemblyName>dotnet-randompackage</AssemblyName>
  </PropertyGroup>
 
  <ItemGroup>
    <Content Include="$(OutputPath)\$(AssemblyName).runtimeconfig.json">
      <Pack>true</Pack>
      <PackagePath>lib\$(TargetFramework)</PackagePath>
    </Content>
  </ItemGroup>
 
  <ItemGroup>
    <PackageReference Include="Microsoft.NETCore.App" Version="1.0.1" />
  </ItemGroup>
 
</Project>
```

**Comment by dsplaisted** [comment]
Thanks for pointing this out, @livarcocc.

In general I wouldn't recommend having `Content` (or other) items that come from your output folder declared in your project file, as it could lead to problems where a target only works correctly if you've previously built (and not cleaned) the project.

However, it would be very hard for people to figure out why their items aren't being included, and if they do figure it out, there's no clean way to override the behavior.  So I've sent #630 to fix this.

**Comment by srivatsn** [comment]
Fixed by #630
